spark2/src/test/java/org/apache/iceberg/spark/source/TestCustomCatalog.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Encoders;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalog.Catalog;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.apache.iceberg.types.Types.NestedField.optional;
+
+public class TestCustomCatalog {
+  private static final String CATALOG_IMPL = String.format("%s.%s.%s", CustomCatalogs.ICEBERG_CATALOG_PREFIX,
+      CustomCatalogs.ICEBERG_DEFAULT_CATALOG, CatalogProperties.CATALOG_IMPL);
+  private static final String WAREHOUSE = String.format("%s.%s.%s", CustomCatalogs.ICEBERG_CATALOG_PREFIX,
+      CustomCatalogs.ICEBERG_DEFAULT_CATALOG, CatalogProperties.WAREHOUSE_LOCATION);
+  private static final String URI_KEY = String.format("%s.%s.%s", CustomCatalogs.ICEBERG_CATALOG_PREFIX,
+      CustomCatalogs.ICEBERG_DEFAULT_CATALOG, CatalogProperties.HIVE_URI);
+  private static final String URI_VAL = "thrift://localhost:12345"; // dummy uri
+  private static final String CATALOG_VAL = "org.apache.iceberg.spark.source.TestCatalog";
+  private static final TableIdentifier TABLE = TableIdentifier.of("default", "table");
+  private static final Schema SCHEMA = new Schema(
+      optional(1, "id", Types.IntegerType.get()),
+      optional(2, "data", Types.StringType.get())
+  );
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  File tableDir = null;
+  String tableLocation = null;
+  HadoopTables tables;
+
+  protected static SparkSession spark = null;
+
+  @BeforeClass
+  public static void startMetastoreAndSpark() {
+    spark = SparkSession.builder().master("local[2]").getOrCreate();
+  }
+
+  @AfterClass
+  public static void stopMetastoreAndSpark() {
+    spark.stop();
+    spark = null;
+  }
+
+  @Before
+  public void setupTable() throws Exception {
+    SparkConf sparkConf = spark.sparkContext().conf();
+    sparkConf.set(
+        String.format("%s.%s", CustomCatalogs.ICEBERG_CATALOG_PREFIX, CustomCatalogs.ICEBERG_DEFAULT_CATALOG),
+        Catalog.class.getName());
+    this.tables = new HadoopTables(spark.sessionState().newHadoopConf());
+    this.tableDir = temp.newFolder();
+    tableDir.delete(); // created by table create
+    this.tableLocation = tableDir.toURI().toString();
+    tables.create(SCHEMA, PartitionSpec.unpartitioned(), String.format("%s/%s", tableLocation, TABLE.name()));
+  }
+
+  @After
+  public void removeTable() {
+    SparkConf sparkConf = spark.sparkContext().conf();
+    sparkConf.remove(CATALOG_IMPL);
+    sparkConf.remove(WAREHOUSE);
+    sparkConf.remove(URI_KEY);
+    tables.dropTable(String.format("%s/%s", tableLocation, TABLE.name()));
+    tableDir.delete();
+    CustomCatalogs.clearCache();
+  }
+
+  @Test
+  public void withSparkOptions() {
```

Comment:
Can you also add a test for a `catalog.db.table` name?

Reply:
sorry, could you explain a bit more?

Reply:
misunderstood your comment. Added a test for that.